torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -150,26 +160,38 @@ class NCCLComm {
   }
 
   ncclComm_t getNcclComm() {
-    std::unique_lock<std::mutex> lock(mutex_);
     if (aborted_) {
-      TORCH_CHECK(false,
-          "NCCL communicator was aborted on rank " + std::to_string(rank_) +
-          ".");
+      std::string abortMsg = "NCCL communicator was aborted on rank " + std::to_string(rank_) + ".";
+      if (commFailureReason_ != c10::nullopt) {
+        abortMsg += " . Original reason for failure was: " + *commFailureReason_;
+      }
+      TORCH_CHECK(false, abortMsg);
```

Comment:
Can we use `fmt::format` here instead?

Separate question - My understanding was that users were seeing errors on the communicator before the work clean up thread actually killed the communicator. But it seems like here we already check for that and used to raise an appropriate exception? Does this mean the failure came probably from other collectives in-flight rather than a new collective call?

Reply:
Yeah, although here with this PR we enhance it with the reason for failure if we try to access a bad communicator.

It's sort of hard to root cause exactly where this failure is coming from as it can stem from several places, but internal logging shows that this happens pretty frequently and the `workCleanupThread` raises this exception (either due to a timeout or an actual NCCL comm. issue).